src/components/schedule-page/new-month-page.component.tsx

```diff
@@ -11,30 +11,47 @@ import { Button } from "../common-components";
 import { useScheduleConverter } from "./import-buttons/hooks/use-schedule-converter";
 import { ScheduleKey } from "../../api/persistance-store.model";
 import { MonthSwitchActionCreator } from "../../state/reducers/month-state/schedule-data/month-switch.action-creator";
+import { LocalStorageProvider } from "../../api/local-storage-provider.model";
+import { MonthDataModel } from "../../common-models/schedule-data.model";
 
 export function NewMonthPlanComponent(): JSX.Element {
   const state = useSelector((state: ApplicationStateModel) => state);
   const currentMonth = state.actualState.persistentSchedule.present.schedule_info.month_number;
   const currentYear = state.actualState.persistentSchedule.present.schedule_info.year;
+  const { revision } = state.actualState;
   const nextDate = getDateWithMonthOffset(currentMonth, currentYear, 1);
   const prevDate = getDateWithMonthOffset(currentMonth, currentYear, -1);
-  const nextMonth =
-    state.history[new ScheduleKey(nextDate.getMonth(), nextDate.getFullYear()).dbKey];
-  const prevMonth =
-    state.history[new ScheduleKey(prevDate.getMonth(), prevDate.getFullYear()).dbKey];
+
+  const [hasValidNext, setHasValidNext] = useState<boolean>(false);
+  const [hasValidPrevious, setHasValidPrevious] = useState<boolean>(false);
   const { monthModel, setSrcFile } = useScheduleConverter();
 
-  const hasValidNext =
-    nextMonth !== undefined && nextMonth.month_info.dates.length > 0 && !nextMonth.isAutoGenerated;
-  const hasValidPrevious =
-    prevMonth !== undefined && prevMonth.month_info.dates.length > 0 && !prevMonth.isAutoGenerated;
+  useEffect(() => {
+    const setNeighbours = async (): Promise<void> => {
+      const storageProvider = new LocalStorageProvider();
+      const nextMonth = await storageProvider.getMonthRevision(
+        new ScheduleKey(nextDate.getMonth(), nextDate.getFullYear()).getRevisionKey(revision)
+      );
+      const prevMonth = await storageProvider.getMonthRevision(
+        new ScheduleKey(prevDate.getMonth(), prevDate.getFullYear()).getRevisionKey(revision)
```

Comment:
Zastawiam się nad tym, czy nie zrobić jednej funkcji, która będzie wygiącać odrazu kilka rewizji. ( na takiej zasadzie, że podajesz jej klucze i ona w takiej samej kolejności zwraca ci elementy z bazy odpowiadające tym kluczom )
 
Bo teraz jest tak, że każda z tych funkcji wyciąga z bazy JSON, tworzy z biego obiekt, przechodzi przez niego i potem koduje to do JSON'a i zapisuje z powrotem. 
Nie jest to pewnie duży narzut, ale dalej narzut.

Reply:
Super uwaga, widać, że się wgyzłeś w ten kod. Faktycznie jest to głupio zrobione, bo my praktycznie zawsze potrzebujemy 3 miesięcy i robimy te 3 pobrania z bazy i filtrowania, a mogło by być tylko 1. 

Nie wydaje się, żeby to było specjalnie duzo roboty, ale wolałbym tego teraz nie wprowadzać, bo mogę coś spieprzyć i dołożyć niepotrzebnie pracy. Jest to jednak optymalizacja, której na razie bardzo niepotrzebujemy, bo działa to dość płynnie. Dlatego na razie poczekam na poprawę testów, później jeśli będą problemy z domergowaniem mojego taska to będę z tym walczył, a jak będzie trochę czasu to wprowadzę jeszcze to usprawnienie. W przeciwnym razie dodajemy to do listy z refactoringiem.